plugins/spreadsheet-view/src/SpreadsheetView/models/ImportWizard.ts

```diff
@@ -134,47 +135,47 @@ export default () => {
       // fetch and parse the file, make a new Spreadsheet model for it,
       // then set the parent to display it
       async import(assemblyName: string) {
-        try {
-          if (!self.fileSource) {
-            return
-          }
+        if (!self.fileSource) {
+          return
+        }
 
-          if (self.loading) {
-            throw new Error('Cannot import, load already in progress')
-          }
+        if (self.loading) {
+          throw new Error('Cannot import, load already in progress')
+        }
 
-          self.selectedAssemblyName = assemblyName
-          self.loading = true
-          const type = self.fileType as keyof typeof fileTypeParsers
-          const typeParser = await fileTypeParsers[type]()
-          if (!typeParser) {
-            throw new Error(`cannot open files of type '${self.fileType}'`)
-          }
+        self.selectedAssemblyName = assemblyName
+        self.loading = true
+        const type = self.fileType as keyof typeof fileTypeParsers
+        const typeParser = await fileTypeParsers[type]()
+        if (!typeParser) {
+          throw new Error(`cannot open files of type '${self.fileType}'`)
+        }
 
-          const { unzip } = await import('@gmod/bgzf-filehandle')
-
-          const filehandle = openLocation(self.fileSource)
-          await filehandle
-            .stat()
-            .then(stat => {
-              if (stat.size > IMPORT_SIZE_LIMIT) {
-                throw new Error(
-                  `File is too big. Tabular files are limited to at most ${(
-                    IMPORT_SIZE_LIMIT / 1000
-                  ).toLocaleString()}kb.`,
-                )
-              }
-            })
-            .then(() => filehandle.readFile())
-            .then(buffer => (self.requiresUnzip ? unzip(buffer) : buffer))
-            .then(buffer => typeParser(buffer as Buffer, self))
-            .then(spreadsheet => {
-              this.setLoaded()
-              getParent(self).displaySpreadsheet(spreadsheet)
-            })
-        } catch (error) {
-          this.setError(error)
+        const { unzip } = await import('@gmod/bgzf-filehandle')
+
+        const filehandle = openLocation(self.fileSource, pluginManager)
+
+        try {
+          await filehandle.stat().then(stat => {
+            if (stat.size > IMPORT_SIZE_LIMIT) {
+              throw new Error(
+                `File is too big. Tabular files are limited to at most ${(
+                  IMPORT_SIZE_LIMIT / 1000
+                ).toLocaleString()}kb.`,
+              )
+            }
+          })
+        } catch (e: any) {
+          // if stat() fails then the filesize cannot be determined and we continue
```

Comment:
possibly just add a console.warn here (and log the actual e object) so that it gets registered